### PR TITLE
Master LPS-157569 | Add test case to assert structured content created with erc by default

### PR DESCRIPTION
**Pull request by yanan**
**Background**
Add a test to assert structured content is created with erc by default and the default value equals to key value

**Test Plan**
Given asset library is created
When with POST request I create a structured Content without a custom erc in asset library
Then structured content is being created and I can see erc value is the key value in the body response

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-157569

### CODE REVIEW

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:relevant

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
Hello @magjed4289 , could you help review when you have time? Thank you!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase [hunk 1]
```diff
@@ -59,11 +59,38 @@ definition {
 				title = "WC WebContent Title");
 		}
 
-		task ("Then structured Content is being createdAnd Then I can see the custom erc in the body response") {
+		task ("Then structured content is being created and I can see the custom erc in the body response") {
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase:62]: the **And** word is reserved for cucumber syntaxis, can you modify the scenario so it does not appear?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase [hunk 2]
```diff
@@ -59,11 +59,38 @@ definition {
...
+			var response = HeadlessWebcontentAPI.createStructuredContentInAssetLibrary(
+				assetLibraryId = "${depotId}",
+				data = "<p>My content</p>",
+				ddmStructureId = "${ddmStructureId}",
+				label = "Text",
+				name = "Text",
+				title = "WC WebContent Title");
+		}
+
+		task ("Then structured content is being created and I can see erc equals to the key in the body response") {
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase:87]: same here, **and** is a reserved word, please modify

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase [hunk 3]
```diff
@@ -59,11 +59,38 @@ definition {
...
+			var defaultErc = HeadlessWebcontentAPI.getKeyFromResponse(responseToParse = "${response}");
+
+			HeadlessWebcontentAPI.assertExternalReferenceCodeWithCorrectValue(
+				expectedactualExternalReferenceCodeValueValue = "${defaultErc}",
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase:91]: `expectedactualExternalReferenceCodeValueValue` - typo
Also, what is the default erc? What two elements are we comparing here?

**Review by magjed4289 — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase (hunk 3 above)
Comment (yanan) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase:91]: When I create a structuredContent without erc, then there will be an erc generated by deault. I've updated the name to "expectedErc" to compare to the actual value.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase (hunk 1 above)
Comment (yanan) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase]: Updated in forced push.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase (hunk 2 above)
Comment (yanan) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase]: Updated in forced push.

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:relevant

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
I will send a new pull request as well as updating the default erc value to uuid value.

**Pull request closed without merging**